ui/src/app/shared/test/testSystem.spec.ts

```diff
@@ -0,0 +1,395 @@
+import { Edge, EdgeConfig } from "../shared";
+import { Role } from "../type/role";
+import { AbstractSystem, Widget } from "./types.spec";
+
+export namespace TestSystem {
+  export class Ems1 extends AbstractSystem {
+    public components: { [id: string]: EdgeConfig.Component; } = {
+      meter0: {
+        alias: "Netzzähler",
+        factoryId: "Meter.Socomec.Threephase",
+        id: "meter0",
+        isEnabled: true,
+        properties: {
+          alias: "Netzzähler",
+          enabled: true,
+          invert: false,
+          modbusUnitId: 5,
+          type: "GRID"
+        },
+        channels: {}
+      }
+    };
+
+    public factories: { [id: string]: EdgeConfig.Factory } = {
+      ['Meter.Socomec.Threephase']:
+      {
+        name: "Meter Socomec Threephase",
+        description: "Implements a threephase Socomec meter. Actual type is identified automatically.",
+        natureIds: [
+          "io.openems.edge.meter.api.ElectricityMeter",
+          "io.openems.edge.bridge.modbus.api.ModbusComponent",
+          "io.openems.edge.common.modbusslave.ModbusSlave",
+          "io.openems.edge.common.component.OpenemsComponent",
+          "io.openems.edge.meter.socomec.threephase.SocomecMeterThreephase",
+          "io.openems.edge.meter.socomec.SocomecMeter"
+        ],
+        properties: [],
+        id: "Meter.Socomec.Threephase",
+        componentIds: [
+          "meter0"
+        ]
+      }
+    };
+
+    public override key: string = "ems1";
+    public widgets: Widget[];
```

Comment:
widgets and keys are unnecessary for now, should i remove them or leave them for later in case we need them

Reply:
It's up to you. From my side its ok to keep them.

Reply:
then lets keep them